ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,87 +82,273 @@ def test_h_max_results():
+                )
+                hillas_dict_CameraFrame[tel_id] = moments_CameraFrame
+                hillas_dict_TelescopeFrame[tel_id] = moments_TelescopeFrame
+
+                event_CameraFrame.dl1.tel[
+                    tel_id
+                ].parameters = ImageParametersContainer()
+                dl1.parameters = ImageParametersContainer()
+
+                event_CameraFrame.dl1.tel[
+                    tel_id
+                ].parameters.hillas = moments_CameraFrame
+                dl1.parameters.hillas = moments_TelescopeFrame
+
+                if (
+                    (moments_CameraFrame.width.value == np.nan)
+                    or (moments_CameraFrame.width.value == np.nan)
+                    or (moments_CameraFrame.width.value == 0)
+                    or (moments_CameraFrame.width.value == 0)
+                ):
+                    event_CameraFrame.dl1.tel[
+                        tel_id
+                    ].parameters.hillas = HillasParametersContainer()
+                    event.dl1.tel[tel_id].parameters.hillas = HillasParametersContainer(
+                        x=float("nan") * u.deg,
+                        y=float("nan") * u.deg,
+                        r=float("nan") * u.deg,
+                        width=float("nan") * u.deg,
+                        length=float("nan") * u.deg,
+                    )
+
+            except HillasParameterizationError as e:
+                print(e)
+                event_CameraFrame.dl1.tel[
+                    tel_id
+                ].parameters.hillas = HillasParametersContainer()
+                event.dl1.tel[tel_id].parameters.hillas = HillasParametersContainer(
+                    x=float("nan") * u.deg,
+                    y=float("nan") * u.deg,
+                    r=float("nan") * u.deg,
+                    width=float("nan") * u.deg,
+                    length=float("nan") * u.deg,
+                )
+
+        if (len(hillas_dict_CameraFrame) < 2) and (len(hillas_dict_TelescopeFrame) < 2):
+            continue
+        else:
+            reconstructed_events += 1
+
+        # last writing of event.dl1 was with telescope frame parameters
+        # make a copy of the event and overwrite its hillas parameters
+
+        for tel_id, hillas in hillas_dict_CameraFrame.items():
+            event_CameraFrame.dl1.tel[tel_id].parameters.hillas = hillas
+
+        fit_CameraFrame(event_CameraFrame)
+        fit_TelescopeFrame(event)
+
+        # Compare old approach with new approach
+        result_CameraFrame = event.dl2.shower["HillasReconstructor"]
+        result_TelescopeFrame = event.dl2.shower["HillasReconstructor"]
+
+        for field in event.dl2.shower["HillasReconstructor"].as_dict():
+            C = np.asarray(result_CameraFrame.as_dict()[field])
+            T = np.asarray(result_TelescopeFrame.as_dict()[field])
+            assert (np.isclose(C, T, rtol=1e-03, atol=1e-03, equal_nan=True)).all()
```

Comment:
which should - if both reconstructions return all NaNs for the same starting event they are giving compatible results

Reply:
I agree, the unit test should only be done on an event that is well reconstructible and then fail if the result is nan

Reply:
ah also the uncertainties of the core and hmax reconstruction are not supported yet and recorded as nan, so we need to compare to nan

Reply:
I filter all events to be well reconstructible or anyway reconstructible enough (width >0 and >=2 images)
I use 10 events and we expect a positive number of reconstructed events (assertion that came from the previous unit-test)
using more than one event make sure that we have compatible reconstruction everytime (since the difference in frame should not depend on the quality of the event to reconstruct)